hedera-state-validator/src/main/java/com/hedera/statevalidation/poc/ChunkedFileIterator.java

```diff
@@ -0,0 +1,251 @@
+import static com.swirlds.merkledb.files.DataFileCommon.FIELD_DATAFILE_ITEMS;
+import static com.swirlds.merkledb.files.DataFileCommon.FIELD_DATAFILE_METADATA;
+
+import com.hedera.pbj.runtime.io.ReadableSequentialData;
+import com.hedera.pbj.runtime.io.buffer.BufferedData;
+import com.hedera.pbj.runtime.io.stream.ReadableStreamingData;
+import com.hedera.statevalidation.poc.ItemData.Type;
+import com.swirlds.merkledb.files.DataFileCommon;
+import com.swirlds.merkledb.files.DataFileMetadata;
+import com.swirlds.merkledb.files.hashmap.Bucket;
+import com.swirlds.merkledb.files.hashmap.ParsedBucket;
+import com.swirlds.virtualmap.datasource.VirtualHashRecord;
+import com.swirlds.virtualmap.datasource.VirtualLeafBytes;
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.Channels;
+import java.nio.channels.FileChannel;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class ChunkedFileIterator implements AutoCloseable {
+    private final FileChannel channel;
+    private final DataFileMetadata metadata;
+
+    private final long startByte;
+    private final long endByte;
+
+    private final ItemData.Type dataType;
+
+    private BufferedInputStream bufferedInputStream;
+    private ReadableSequentialData in;
+    private BufferedData dataItemBuffer;
+    private long currentDataItemFilePosition;
+    private boolean closed = false;
+
+    private long boundaryOffset = 0L;
+
+    public ChunkedFileIterator(
+            Path path, DataFileMetadata metadata, Type dataType, long startByte, long endByte,
+            AtomicLong totalBoundarySearchMillis)
+            throws IOException {
+        this.channel = FileChannel.open(path, StandardOpenOption.READ);
+        this.metadata = metadata;
+
+        this.startByte = startByte;
+        this.endByte = endByte;
+
+        this.dataType = dataType;
+
+        if (startByte > 0) {
+            // Find boundary, then position channel and open streams
+            long startTime = System.currentTimeMillis();
+            this.boundaryOffset = findBoundaryOffset();
+            long boundaryOffsetSearchTime = System.currentTimeMillis() - startTime;
+//            System.out.println("Found boundary offset in:" + boundaryOffsetSearchTime + " ms");
+            totalBoundarySearchMillis.addAndGet(boundaryOffsetSearchTime);
+            channel.position(startByte + boundaryOffset);
+            openStreams();
+        } else {
+            // At file start
+            channel.position(startByte);
+            openStreams();
+        }
+    }
+
+    private void openStreams() {
+        var channelStream = Channels.newInputStream(channel);
+        // 16777216 took from MerkleDbConfig.iteratorInputBufferBytes
+        this.bufferedInputStream = new BufferedInputStream(channelStream, 16777216);
```

Comment:
16M buffer looks redundant. Most our data items don't exceed a few kilobytes, 128K or 256K buffer should be enough